packages/react-query/src/internals/getQueryKey.ts

```diff
@@ -1,11 +1,51 @@
+import { AnyProcedure, AnyRouter, inferProcedureInput } from '@trpc/server';
+import { QueryType, getArrayQueryKey } from '../internals/getArrayQueryKey';
+import { DecorateProcedure, DecoratedProcedureRecord } from '../shared';
+
 /**
  * We treat `undefined` as an input the same as omitting an `input`
  * https://github.com/trpc/trpc/issues/2290
  */
-export function getQueryKey(
+export function getQueryKeyInternal(
   path: string,
   input: unknown,
 ): [string] | [string, unknown] {
   if (path.length) return input === undefined ? [path] : [path, input];
   return [] as unknown as [string];
 }
+
+/**
+ * Method to extract the query key for a procedure
+ * @param type - defaults to `any`
+ * @link https://trpc.io/docs/useContext#-the-function-i-want-isnt-here
+ */
+export function getQueryKey<
+  TProcedureOrRouter extends AnyProcedure | AnyRouter,
```

Comment:
As this now accepts any procedure (queries and mutations (even subskriptions which we might not want to sccept) maybe we want some tests for mutations too? 

I think it should `just work` but good to have some validation.

This would also close #3528

Reply:
cool i added a test for mutations

Reply:
https://youtube.com/shorts/uzgwQFPMuG4?feature=share